ethtool: Add support of ethtool feature

cathay4t opened the pull request:
Example:

```yaml interfaces:
- name: eth1
type: ethernet
state: up
ethtool:
  feature:
    highdma: true
    rx-checksum: true
    rx-gro: true
```

Nmstate only shows changeable features via `libnmstate.show()`.

Unlike the ethtool CLI tool, nmstate show using feature name provided by
kernel instead of alias. When applying state, nmstate support desire state
to use these alias which pre-exists in ethtool CLI tool:
* `rx`
* `rx-checksumming`
* `ufo`
* `gso`
* `generic-segmentation-offload`
* `gro`
* `generic-receive-offload`
* `lro`
* `large-receive-offload`
* `rxvlan`
* `rx-vlan-offload`
* `txvlan`
* `tx-vlan-offload`
* `ntuple`
* `ntuple-filters`
* `rxhash`
* `receive-hashing`

The NetworkManager itself also has some alias, the nm plugin is taking care
of it.

Integration test case added testing both this alias processing and setting
`rx-checksum` value.

Unit test case included to test the alias processing.

#

ffmancera commented on libnmstate/ifaces/ethtool.py:
```diff
@@ -71,28 +80,70 @@ def rx(self):
+                self._info.pop(key)
+                original_desire.pop(key, None)
+
+    def items(self):
+        for k, v in self._info.items():
+            yield k, v
```
I do not like this name, it could be confusing with the dictionary function `items()`. But I do not have a better option.

cathay4t replied:
With this name, we don't even need a comment to explain what this function do. And it is actually doing the same thing as `items()` of dictionary does.

ffmancera commented on libnmstate/nm/ethtool.py:
```diff
@@ -60,4 +59,38 @@ def create_ethtool_setting(iface_ethtool, base_con_profile):
+    if NM.ethtool_optname_is_feature(nm_feature_name):
+        nm_setting.option_set_boolean(nm_feature_name, value)
+    else:
+        logging.warn(
+            f"Ethtool feature {kernel_feature_name} is invalid "
+            "or not supported by current NetworkManager"
+        )
```
This should be a warning or an error? What would happen? Nmstate will fail with a non-userfriendly error from NM, right?

cathay4t replied:
User will get a verification error for this.

cathay4t replied:
Let me raise exception herre

cathay4t replied:
New patch uploaded. Raise `NmstateValueError` for invalid option. Also attached integration test case for this.

ffmancera reviewed (changes requested):
I have a few questions. Other than that it looks good to me.

cathay4t committed:
ethtool: Add support of ethtool feature

Example:

```yaml
interfaces:
- name: eth1
  type: ethernet
  state: up
  ethtool:
    feature:
      highdma: true
      rx-checksum: true
      rx-gro: true
```

Unlike the ethtool CLI tool, nmstate show using feature name provided by
kernel instead of alias. When applying state, nmstate support desire
state to use these alias which pre-exists in ethtool CLI tool:
 * `rx`
 * `rx-checksumming`
 * `ufo`
 * `gso`
 * `generic-segmentation-offload`
 * `gro`
 * `generic-receive-offload`
 * `lro`
 * `large-receive-offload`
 * `rxvlan`
 * `rx-vlan-offload`
 * `txvlan`
 * `tx-vlan-offload`
 * `ntuple`
 * `ntuple-filters`
 * `rxhash`
 * `receive-hashing`

The NetworkManager itself also has some alias, the nm plugin is taking
care of it.

Nmstate only shows changeable features via `libnmstate.show()`.

Integration test case added testing both this alias processing and
setting `rx-checksum` value.

Unit test case included to test the alias processing.

Signed-off-by: Gris Ge <fge@redhat.com>

ffmancera reviewed (approved):
LGTM. Thanks!

cathay4t committed:
nispor ovs: Hide the `ovs-system` kernel nic

The

Signed-off-by: Gris Ge <fge@redhat.com>
